### ISSUE TITLE
Possible restructuring of code for generating train, test data

### ISSUE THREAD

**Issue by anubhabghosh** [anubhabghosh/NovelESN#4]
- Possibly trying to change the code for generating training, testing datasets. So I (@anubhabghosh ) will not try to modify the `main.py` too much. 

- @BarKetPlace If you need, I can add some comments to the `get_train_test_dynamo.py`, `get_train_test_realsolar.py`, so that it is more clear on how the current train test splits occur.

**Comment by anubhabghosh** [comment]
For reference, the current algorithm for prediction applied for models like `linear_ar`:

For predicting let's say *solar cycle* no. 24:

- We have past 23 solar cycles of data (N_samples x 1) that includes training + validation data.  The data for solar cycle 24 is test data.

- Depending on the **input** dimension of the model, the training + validation dataset is *restructured* as (N_samples x input_dimension) (this input dimension is denoting the number of values taken by the model to predict one single future value). This **input** dimension is decided by `num_taps` parameter for `linear_ar` model.

- In the train ( ), the training + validation dataset is split such that approximately 75% of the 23 cycles are chosen for training, and the remaining for validation.

- Once, the model is trained, the **last row** in training + validation dataset (or effectively the validation dataset, as the first 75% cycles are used for training), is fed as input to the model. So, the test input is of dimension ( 1 x input_dimension )

- The model predicts a **single** future value, then uses this value to make a *new* test input of dimension (1 x input_dimension).  Then it predicts the next future prediction, and iteratively this process is continued until we get the desired number of points for cycle 24.

**Current issue:**

- We are using the last entry of the validation set as the starting entry for the test set (which is technically correct), but this would have been better if this was the training set perhaps. Not sure, if there's a better way to form a validation set though.
